test/KitPidHelpers.cpp

```diff
@@ -120,10 +122,60 @@ void helpers::waitForKitPidsReady(
     waitForKitProcessCount(testname, 0, 1, timeoutMs, retryMs);
 }
 
-void helpers::waitForKitPidsKilled(
-        const std::string& testname,
-        const std::chrono::milliseconds timeoutMs /* = KIT_PID_TIMEOUT_MS */,
-        const std::chrono::milliseconds retryMs /* = KIT_PID_RETRY_MS */)
+void helpers::killPid(const std::string& testname, const pid_t pid)
+{
+    TST_LOG("Killing " << pid);
+    if (kill(pid, SIGKILL) == -1)
+        TST_LOG("kill(" << pid << ", SIGKILL) failed: " <<
+                Util::symbolicErrno(errno) << ": " << std::strerror(errno));
+}
+
+namespace {
+    std::set<pid_t> intersect(const std::set<pid_t> &a,
+                              const std::set<pid_t> &b)
+    {
+        std::set<pid_t> result;
+        for (auto pid : a)
+        {
+            if (b.find(pid) != b.end())
+                result.insert(pid);
+        }
+        return result;
+    }
+}
+
+void helpers::killAllKitProcesses(
+    const std::string& testname,
+    const std::chrono::milliseconds timeoutMs /* = COMMAND_TIMEOUT_MS * 8 */,
+    const std::chrono::milliseconds retryMs /* = 10ms */)
 {
-    waitForKitProcessCount(testname, 0, 0, timeoutMs, retryMs);
+    auto before = getKitPids();
+    for (pid_t pid : before)
+        killPid(testname, pid);
+
+    TST_LOG("Waiting for these kit processes to close: " << getPidList(before));
+
+    bool pass = false;
+    int tries = (timeoutMs / retryMs);
+
+    std::set<pid_t> inters;
+    while (tries >= 0 && !pass)
+    {
+        std::this_thread::sleep_for(retryMs);
+
+        auto current = getKitPids();
+
+        inters = intersect(current, before);
+        pass = inters.empty();
+        tries--;
+
+        TST_LOG("Current kit processes: " << getPidList(current));
+    }
+
+    TST_LOG("Finished waiting for all previous kit processes to close"
+            << " before: " << getPidList(before)
+            << " current: " << getPidList(getKitPids())
+            << " intersection: " << getPidList(inters));
+
+    LOK_ASSERT("Timed out waiting for these kit processes to close" && pass);
```

Comment:
Use `LOK_ASSERT_MESSAGE_SILENT("message",pass)`

This will hide confusing "Timed out" messages from the log in the passing case where it didn't time out.

Or use `if(pass) { TST_LOG() } else { LOK_ASSERT_FAIL() }`, like in waitForKitProcessCount, so you can also customize the other output for passing/failing, and skip logging the intersection which will always be empty for the passing case.